### PR TITLE
Feature/#376 color variable naming structure

### LINKED ISSUE #376: Context Based Color Variables
I have been very familiar with context based color variables which makes it easier to read and understand code. Right now, the color variables are defined with a color name. What are your thoughts on changing it to a context-based naming scheme e.g. $primary-color, $secondary-color, $button-color, $link-color?

I have seen the context-based scheme used within the other areas of SCSS but defining them in one place, in my opinion will make it very easy to follow.

### PR DESCRIPTION
**Pull request by jomurgel**
Closes #376

### DESCRIPTION ###
- Removed color assignments from `_color-vars` formerly `_colors` in "variables" into `_color-assignments` partial to keep all additional global color variables separate.
- Moves all color variables into `_color-assignments` partial and ensures that not color variables eg. `$color-white` is used anywhere outside of the `_color-vars` or `_color-assignments` partials.
- Renamed partials for variables `prefix-vars.scss` to avoid having any scss partials with the same name to avoid confusion.
- Moved `_sprits.scss` into mixins.
- Added a host of new color variables for forms, inputs, buttons, etc. A new variable anywhere a color variable was used outside of `_color-vars` formerly `_colors` inside variables.
- Adds additional documentation and `sassdocs` support.
- General cleanup and linting.

### CODE REVIEW

**Comment by mharis** [comment]
@jomurgel My suggestion would be to keep color variables and assignments in the same file so that devs can stay in one file, easier to navigate and remember. This is just me being lazy.

**Review by gregrickaby — APPROVED** [review]
I think there's some truth to @mharis's suggestion. I'll leave that up to the group though.

This is amazing, @jomurgel! 🌮 👍

**Review by gregrickaby — DISMISSED** [review]
@jomurgel Can you merge in `master` and recompile? Thanks!

**Comment by coreymcollins** [comment]
I think I tend to agree with keeping the color variables and assignments all in one file. The less I have to poke around to find the place where a color is set, the better. The partial will wind up being long, but I think it would be nicer to have all of that stuff in one place.

Awesome work on this, Jo. This was a lot to take on and it looks good so far! Going to wait to run through a full review until after the merge conflict items are taken care of, but so far this is amazing.

Thanks!

**Comment by jomurgel** [comment]
@mharis @coreymcollins I consolidated the `_color-assignments.scss` and `_color-var.scss` into one partial, updated the TOC on that part and did some general cleanup.

Passing it back for another review 👍

File: assets/sass/base/_scaffolding.scss
Comment (coreymcollins) [inline assets/sass/base/_scaffolding.scss]: Borders should use `px` instead of `rem()`

File: assets/sass/base/_scaffolding.scss
Comment (coreymcollins) [inline assets/sass/base/_scaffolding.scss]: Borders should use `px` instead of `rem()`

File: assets/sass/modules/_content.scss
Comment (coreymcollins) [inline assets/sass/modules/_content.scss:77]: I know this is already in the theme, but I think it might be best to remove this and instead add the `screen-reader-text` class and ARIA labels to the entry titles of posts where we want to hide the title. Right now, the post title is hidden so on an archive page I don't have any context about the post when the screen reader gets to it.

But maybe that's intentional? I'm not sure why these were hidden in the first place. 🤔

File: assets/sass/modules/navigation/_mobile-navigation.scss
Comment (coreymcollins) [inline assets/sass/modules/navigation/_mobile-navigation.scss]: We're setting `outline` with `rem()` here and then `px` later on in the file. We don't have a specification in our Units of Measurement graphic... do we need an update to a new version after over a year at v0.3.0?! 😮

File: assets/sass/utilities/variables/_color-var.scss
Comment (coreymcollins) [inline assets/sass/utilities/variables/_color-var.scss:1]: Some general comments here, rather than noting in a handful of comments on individual lines.

Sometimes the color variables are set like `$color-background-element` and other times `$color-element-background`. Sometimes variables use `color` twice, like `$color-element-color`.

I wonder if we can make this more consistent so everything is always `$color-background-element` for backgrounds and `$color-element` for colors?

Some examples from in this file:
`$color-background-screen-reader`
`$color-text-screen-reader`

I like those!

Then we have some others like this:
`$color-icon-close-background`
`$color-modal-close-color`

Those feel a little clunkier to me, especially the second one where `color` is used twice.

More than happy to compromise here if these were done a specific way for specific reasons, but just something I noticed that didn't seem to be done consistently.

File: assets/sass/utilities/variables/_modal-var.scss
Comment (coreymcollins) [inline assets/sass/utilities/variables/_modal-var.scss]: Border radius should be `px` according to our UoM graphic. Do we need to update the theme or update the graphic?

File: assets/sass/utilities/variables/_transition-var.scss
Comment (coreymcollins) [inline assets/sass/utilities/variables/_transition-var.scss]: Can we call a transition variable from a transition variable? For instance, instead of doing what we have now do `$transition-hamburger: $transition-transform, $transition-bg-color;`?

Since we're not customizing the speed or easing, I'm wondering if that could work without Gulp getting weird about it.

**Review by coreymcollins — CHANGES_REQUESTED** [review]
The amount of work here is staggering. You rule for taking this on!

I have some comments on various partials here and there. Nothing too crazy – some questions/thoughts in some partials and some small unit of measurement tweaks to be made.

I think having these variables all in one partial is going to be awesome, and I love how many different variables are set for colors now with one place to find and manage them. Awesome work, Jo!

File: assets/sass/modules/_content.scss
Comment (jomurgel) [reply assets/sass/modules/_content.scss:77]: That's a good question. I'm going to ignore this comment in this branch and open a new issue #429

File: assets/sass/utilities/variables/_color-var.scss
Comment (jomurgel) [reply assets/sass/utilities/variables/_color-var.scss:1]: Good call. Should keep these consistent for sure.

File: assets/sass/utilities/variables/_transition-var.scss
Comment (jomurgel) [reply assets/sass/utilities/variables/_transition-var.scss]: This would work, yeah, as long as the one being called is defined before the other. ie, `$transition-transform` is defined before `$transition-bg-color`

**Comment by jomurgel** [comment]
@coreymcollins I made several updates.

- Most notably converting all of the color variables to the `$color-type-element` or `$color-element` format throughout. 
- Opened up #429 to talk about those SRT classes and ARIA labels.
- Updated rem to px in a few locations.
- Updated the transitions partial to utilize variables for the transition names (with the exception of all which doesn't need to be reused).

**Comment by coreymcollins** [comment]
Awesome work here, Jo!

I pushed up one commit here:
https://github.com/WebDevStudios/wd_s/pull/382/commits/0e03ba4836a745f5fa0c8d0093ddc58b1d1a8f76

A couple of the variables set for transitions were missing the `$ease` variable when multiple transitions were being used.

Other than that, this looks really great.

I'd love to have a second pair of eyes give this the OK as well since there is a *_lot_* to look at.

Once it's approved by two of us, `master` will need to be merged in to resolve any conflicts there as we've had a pretty active day of merging things in so I'm sure there will be some conflicts.

Thanks again for your hard work on this!

**Review by coreymcollins — APPROVED** [review]

**Comment by jomurgel** [comment]
@gregrickaby @coreymcollins I've merged in `master` and recompiled all the things. In doing so I also fixed a couple transition variables, added a transition variable for the accordion and fixed a couple of variable names that had since been updated.

**Comment by mharis** [comment]
I did a review @coreymcollins and this seems good overall.

One note would that maybe we can move
```
/// Global input padding.
$padding-input: rem(6);
``` 

From color variables file to the `assets/sass/modules/_forms.scss` file.

**Comment by jomurgel** [comment]
I ended up removing this variable altogether and hardcoding the value in the `_forms.scss`, `_gravity-forms.scss`, and `_buttons.scss` respectively.

Agreed that that should be moved from the `_color-var.scss` partial as it's not a color var, but since these elements are always tweaked independently of one-another I don't think it makes much sense to have them using the same variable.

**Comment by jomurgel** [comment]
Alrighty. I ran through this PR again. I think it's as clean and purposeful as it can be until we actually give some purposeful design elements to the default WD_s. Just need another pair of eyes from @coreymcollins, @gregrickaby, and @ImBigWill.

**Review by ImBigWill — APPROVED** [review]

**Comment by jomurgel** [comment]
@gregrickaby I've merged in and fixed issue getting through buddy. Should be gtg here. 👍

**Comment by coreymcollins** [comment]
I ran a final Sass Lint against this branch and found only two issues, for which I have pushed up fixes
https://github.com/WebDevStudios/wd_s/pull/382/commits/6483eea01a4f592ea8793bedfea836c748c8c687:
- https://github.com/WebDevStudios/wd_s/pull/382/commits/4307679a26bdf042844bcb9e7ee7ef6ef4d57f7e
- https://github.com/WebDevStudios/wd_s/pull/382/commits/6483eea01a4f592ea8793bedfea836c748c8c687

Greg's main request previously was to resolve merge conflicts, which was done, and Haris, Will, and myself have reviewed the branch. I think we're looking good here!

Going to dismiss Greg's lingering review (as he's hanging out with Donald Duck right now, the lucky son of a...) and merge this puppy in.

Thanks!

**Pull request merged**
